libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt

```
@@ -339,65 +341,17 @@ internal class MapboxTripSession(
-                return
-            }
-
-            var triggerObserver = false
-            if (tripStatus.navigationStatus.routeState != RouteState.INVALID) {
-                val nativeBannerInstruction = tripStatus.navigationStatus.bannerInstruction
-                val bannerInstructions =
-                    tripStatus.navigationStatus.getCurrentBannerInstructions(primaryRoute)
-                triggerObserver = bannerInstructionEvent.isOccurring(
-                    bannerInstructions,
-                    nativeBannerInstruction?.index
-                )
-            }
-            val remainingWaypoints = tripStatus.calculateRemainingWaypoints()
-            val latestBannerInstructionsWrapper = bannerInstructionEvent.latestInstructionWrapper
-            val routeProgress = getRouteProgressFrom(
-                tripStatus.route,
-                tripStatus.navigationStatus,
-                remainingWaypoints,
-                latestBannerInstructionsWrapper?.latestBannerInstructions,
-                latestBannerInstructionsWrapper?.latestInstructionIndex,
-                lastVoiceInstruction
-            ).also {
-                if (it == null) {
-                    logD(
-                        "route progress update dropped - " +
-                            "currentPrimaryRoute ID: ${primaryRoute?.id}; " +
-                            "currentState: ${status.routeState}",
-                        LOG_CATEGORY
-                    )
+            try {
+                processNativeStatus(status)
+            } catch (error: Throwable) {
+                logE(LOG_CATEGORY) {
+                    "Error processing native status update: origin=$origin, status=$status.\n" +
+                        "Error: $error\n" +
+                        "MapboxTripSession state: " +
+                        "isUpdatingRoute=$isUpdatingRoute, primaryRoute=${primaryRoute?.id}"
```

I'm wondering, does it make sense to move this text to exception message and don't log it manually 🤔

---

I think makes sense to leave it as a log message in case if the exception will be swallowed in the upstream code.